[WFLY-10288] Remove remaining Java EE 8 Preview information

jamezp opened the pull request:
https://issues.jboss.org/browse/WFLY-10288

The first commit just updates the documentation removing the reference to Java EE 8 preview mode.

The second commit removes the JAX-WS Java EE 7 support. It looks like this one got missed somehow.

The third commit moves the `BeanValidationEE8TestCase` to the integration tests.

The forth commit cleans up the rest of the Java EE 8 tech preview resources.

jamezp committed:
[WFLY-10288] Update the documentation for Java EE 8 and remove the Java EE 8 preview references.

jamezp committed:
[WFLY-10288] Remove jboss-jaxws-api_2.2 and move to jboss-jaxws-api_2.3 for Java EE 8.

jamezp committed:
[WFLY-10288] Remove Java EE 8 temporary tests and the standalone-ee8.xml template. Remove other references to the ee8.preview.mode.

bstansberry commented on docs/pom.xml:
```diff
@@ -48,6 +48,10 @@
                     <configuration>
+                        <attributes>
+                            <javaee_version>8</javaee_version>
+                            <wildflyversion>${verifier.product.release.version}</wildflyversion>
```
This is kind of a hijack of something meant and documented for a different purpose. If we want different uses we should probably rework the property and clarify it's use.

For the verifier use it's major.minor while wildflyversion in the past was just major. So there's a bit of a mismatch there.

bstansberry replied:
Any change to the properties should be discussed with Yeray though, as these are important for eap-ization and any changes will need to be accounted for there.

jamezp replied:
Yeah. It's on my documentation todo list to change several constants to attributes.

Sure yeah. We don't have to use that version as it was set to just "13" before. I just figured the less places we need to make changes the better. I kind of wonder if these docs should just be in their own repository anyway.

kabir replied:
@jamezp I think a product.docs.server.version in the root pom next to verifier.product.release.version will make this cleaner

jamezp replied:
Done.

scottmarlow commented:
Would it make sense to keep the BeanValidationEE8TestCase test?  Perhaps move it to testsuite/integration/basic?  I'm not sure if there is anything worth keeping in it but thought I would ask.

asoldano reviewed (approved):
Thanks for the jaxws module fix

jamezp commented:
We could keep the test. I just deleted the `ee8-temp` module. Since it was in that module I guess it fell victim :)

rsvoboda commented:
@mjurc - keep the BeanValidationEE8TestCase ?

mjurc commented:
Please move the `BeanValidationEE8TestCase` to `testsuite/integration/basic` module. It seems to be a bit more thorough than our standing `BeanValidationTestCase`.

kabir reviewed (changes requested):
In addition to the above comments, please also remove the full-feature-pack-licenses-addendum.html and servlet-feature-pack-licenses-addendum.html files as I think we should now have removed all the remaining hardcoded versioned jars (ie the ones not coming from the pom)?

jamezp committed:
[WFLY-10288] Move the BeanValidationEE8TestCase to the integration tests.

jamezp commented:
Thanks @kabir I missed those and they are now removed.

@mjurc @rsvoboda I added a commit that moves the `BeanValidationEE8TestCase` to the integration tests.

rsvoboda reviewed (approved):
I have no further comments?

jamezp commented:
@scottmarlow It was moved in commit https://github.com/wildfly/wildfly/pull/11424/commits/5ddf0bfc13382cb580cd1c4b8f100467c34fc972. It's the third commit.
